api/src/main/java/org/openmrs/api/impl/EncounterServiceImpl.java

```diff
@@ -217,17 +218,51 @@ public Encounter saveEncounter(Encounter encounter) throws APIException {
 				toAdd.add(newObs);
 			}
 		}
-		
-		for (Obs o : toRemove) {
+
+		// remove all obs in toRemove recursively from encounter
+		removeAllWithGroupMembers(encounter, toRemove);
```

Comment:
@mddubey please dont comment your code. you write one line of comment for one line of code. your method is private, you can easily give it a more descriptive name if you feel the need to explain what it does. the method name/signature gives you a chance to explain.